dashboard/test/factories/pd_factories.rb

```diff
@@ -452,11 +452,16 @@
           create :pd_attendance, session: session, teacher: teacher
         end
       end
-      if evaluator.cdo_scholarship_recipient
+
+      scholarship_params = {
+        user: teacher,
+        course: Pd::Workshop::COURSE_KEY_MAP[evaluator.workshop.course],
+        application_year: evaluator.workshop.school_year
+      }
+
+      if evaluator.cdo_scholarship_recipient && Pd::ScholarshipInfo.find_by(scholarship_params).nil?
```

Comment:
can you add a comment as to why the second check is needed?

Reply:
Done! Take a look and LMK if unclear.

Reply:
looks good, thanks!